salt/modules/junos.py

```diff
@@ -150,75 +150,48 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
             log.debug("Deleted cached folder: %s", self._cached_folder)
 
 
-def _timeout_decorator(function):
-    @wraps(function)
-    def wrapper(*args, **kwargs):
-        if "dev_timeout" in kwargs or "timeout" in kwargs:
-            ldev_timeout = max(kwargs.pop("dev_timeout", 0), kwargs.pop("timeout", 0))
-            conn = __proxy__["junos.conn"]()
-            restore_timeout = conn.timeout
-            conn.timeout = ldev_timeout
-            try:
-                result = function(*args, **kwargs)
-                conn.timeout = restore_timeout
-                return result
-            except Exception:  # pylint: disable=broad-except
-                conn.timeout = restore_timeout
-                raise
-        else:
-            return function(*args, **kwargs)
-
-    return wrapper
-
-
 def _timeout_decorator_cleankwargs(function):
+    """
+    Removing __pub kwargs during run and adding it back after to ensure non exposure.
+    Setting the timeout for the connection during run and then resetting the timeout.
+    Timeout has to be re-added for proxys running on the actual juniper
+    """
+
     @wraps(function)
     def wrapper(*args, **kwargs):
+        timeout_changed = False
+        old_kwargs = kwargs
```

Comment:
You actually need to create a copy, most likely, even a `copy.deepcopy` of the dictionary, otherwise changing `kwargs` will also change `old_kwargs`